pkg/chunk/series_store.go

```diff
@@ -298,6 +351,38 @@ func (c *seriesStore) lookupChunksBySeries(ctx context.Context, from, through mo
 	return result, err
 }
 
+func (c *seriesStore) lookupSingleChunkBySeries(ctx context.Context, from, through model.Time, seriesIDs []string) ([]string, error) {
+	log, ctx := spanlogger.New(ctx, "SeriesStore.lookupSingleChunkBySeries")
+	defer log.Span.Finish()
+
+	userID, err := user.ExtractOrgID(ctx)
+	if err != nil {
+		return nil, err
+	}
+	level.Debug(log).Log("seriesIDs", len(seriesIDs))
+	var result []string
+	for _, seriesID := range seriesIDs {
+		qs, err := c.schema.GetChunksForSeries(from, through, userID, []byte(seriesID))
+		if err != nil {
+			return nil, err
+		}
+		if len(qs) > 0 {
+			entries, err := c.lookupEntriesByQueries(ctx, []IndexQuery{qs[0]})
```

Comment:
This is not valid, you should get all the entries (chunks) using all the queries and then use the first chunk. This is because if the store can return no chunks on the first query but chunks on the second one.

Further, I think the selection of which chunk to load should happen here: https://github.com/cortexproject/cortex/pull/1346/files#diff-1d35ed616880c655543a534fed562675R198 after `filterChunksByTime`.

Reply:
Yeah I was wondering about chunk missing too, ok let's try another approach thanks.